sympy/functions/elementary/integers.py

```diff
@@ -243,15 +286,46 @@ def _eval_Eq(self, other):
     def __lt__(self, other):
         if self.args[0] == other and other.is_real:
             return S.false
+        if other is S.Infinity and self.is_finite:
+            return S.true
+
+        if S(other).is_integer:
+            return self.args[0] <= other - 1
+
         return Lt(self, other, evaluate=False)
 
+    def __gt__(self, other):
+        if self.args[0] == other and other.is_real:
+            return S.false
+        if other is S.NegativeInfinity and self.is_finite:
+            return S.true
+
+        if S(other).is_integer:
+            return self.args[0] > other
+
+        return Gt(self, other, evaluate=False)
+
     def __ge__(self, other):
         if self.args[0] == other and other.is_real:
             return S.true
-        if other is S.NegativeInfinity and self.is_real:
+        if other is S.NegativeInfinity and self.is_finite:
             return S.true
+
+        if S(other).is_integer:
+            return self.args[0] > other - 1
+
         return Ge(self, other, evaluate=False)
 
+    def __le__(self, other):
+        if self.args[0] == other and other.is_real:
+            return S.false
```

Comment:
return S.true

Reply:
It is false- ceiling(5.5) is not <= 5.5.

Reply:
The problem is I need to move the integer case to be first.

Reply:
I thought is was the floor function. My bad.